todoist/dashboard/subpages/log_viewer.py

```diff
@@ -0,0 +1,211 @@
+    ]
+    
+    for log_dir in common_log_dirs:
+        if log_dir.exists() and log_dir.is_dir():
+            for log_file in log_dir.glob("*.log"):
+                if log_file.is_file():
+                    log_files.append(str(log_file))
+    
+    return sorted(list(set(log_files)))  # Remove duplicates and sort
+
+
+def read_log_file(file_path: str, tail_lines: int = 40, page: int = 1) -> tuple[Optional[str], int]:
+    """Read lines from a log file with pagination support.
+    
+    Args:
+        file_path: Path to the log file
+        tail_lines: Number of lines per page
+        page: Page number (1-based)
+        
+    Returns:
+        Tuple of (content, total_pages)
+    """
+    try:
+        with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
+            lines = f.readlines()
+            
+        total_lines = len(lines)
+        total_pages = max(1, (total_lines + tail_lines - 1) // tail_lines)  # Ceiling division
+        
+        if page < 1:
+            page = 1
+        elif page > total_pages:
+            page = total_pages
+            
+        # Calculate start and end indices for the requested page
+        # Page 1 shows the most recent lines (end of file)
+        # Page 2 shows the lines before that, etc.
+        end_line = total_lines - (page - 1) * tail_lines
+        start_line = max(0, end_line - tail_lines)
+        
+        if start_line < 0:
+            start_line = 0
+        if end_line > total_lines:
+            end_line = total_lines
+            
+        page_lines = lines[start_line:end_line]
+        return ''.join(page_lines), total_pages
+        
+    except Exception as e:
+        return f"Error reading file: {str(e)}", 1
+
+
+def render_log_viewer_page() -> None:
+    """Render the log viewer page."""
+    st.title("📋 Log Viewer")
+    st.write("View and monitor log files from your Todoist Assistant automations and other processes.")
+    
+    # Get available log files
+    log_files = get_available_log_files()
+    
+    if not log_files:
+        st.info("No .log files found in the current directory or common log locations.")
+        st.write("Log files will appear here once automations are run or other processes create log files.")
+        return
+    
+    # Display available log files
+    st.subheader("Available Log Files")
+    
+    # File selection
+    selected_file = st.selectbox(
+        "Select a log file to view:",
+        options=log_files,
+        format_func=lambda x: os.path.basename(x)
+    )
+    
+    if selected_file:
+        # Display file information
+        file_path = Path(selected_file)
+        file_stat = file_path.stat()
+        
+        col1, col2, col3 = st.columns(3)
+        with col1:
+            st.metric("File Size", f"{file_stat.st_size / 1024:.1f} KB")
+        with col2:
+            st.metric("Modified", file_stat.st_mtime)
+        with col3:
+            st.metric("Location", str(file_path.parent))
```

Comment:
location should start from `./` as repository root, there is no need for a absolute path from root of filesystem.